bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -117,12 +145,17 @@ protected void removeSitemapProvider(SitemapProvider provider) {
      * Creates a new subscription with the given id.
      *
      * @param callback an instance that should receive the events
-     * @returns a unique id that identifies the subscription
+     * @returns a unique id that identifies the subscription or null if the limit of subscriptions is already reached
      */
     public String createSubscription(SitemapSubscriptionCallback callback) {
+        if (callbacks.size() >= maxSubscriptions) {
```

Comment:
If we would use

```Java
if (maxSubscriptions > 0 && callbacks.size() >= maxSubscriptions) {
```

it will allow the user to use a negative number for unlimited subscriptions.
I don't know if this makes sense.
The "user" could also use the maximal integer to get that behavior.

But if we want to keep the behavior as it has been before the default could also point to a negative number.

I don't know if unlimited needs to be supported at all, but let's think about it.

Reply:
Good idea, a negative value could be interpreted as unlimited for advanced users.
I want to keep 0 as possible limit for users having problems with SSE.
As our goal is to avoid memory problem, I think the default should define a reasonable limit that should never be reached in normal conditions.